pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -44,24 +44,28 @@ import (
 	tracertestutil "github.com/DataDog/datadog-agent/pkg/network/tracer/testutil"
 	"github.com/DataDog/datadog-agent/pkg/network/tracer/testutil/grpc"
 	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-func httpSupported(t *testing.T) bool {
+func httpSupported() bool {
 	currKernelVersion, err := kernel.HostVersion()
-	require.NoError(t, err)
+	if err != nil {
```

Comment:
so you're right, but currently there's inconsistency in the tests
for example tests that call `HTTPSSupported` won't fail, but only warn and skipped

Reply:
and regardless, I don't think that's the correct place to get such a failure
we should have instead a UT for that method, and there a test should fail if we cannot call the method

Reply:
Sorry to me UT = University of Texas :-) can you explain?

Reply:
Lol
Meant Unit Test

Reply:
We could do that but then it could fail at other times during the full ci run, e.g. I think we should just make sure we know why the test was skipped or failed, with the current changes it is isn't clear we can see that in the ci logs.

Reply:
couple of more locations where tests can be skipped due to the same issue:
1. every tests that uses `isTestIPv6Enabled`.
2. `TestUDPPythonReusePort`
3. `TestUDPReusePort`
4. `TestSendfileRegression` (UDP)

Reply:
Instead of getting the kernel version in httpSupported, I'm now reusing `kv` which is defined in `tracer_linux_test.go` and being used by all the methods and tests I mentioned in the previous comment.

Alongside, I added a check in TestMain, to ensure we're able to get the kernel version without error.
On error I'm failing the tests (via TestMain)